examples/account/key_exchange.rs

```diff
@@ -0,0 +1,119 @@
+
+  // Sets the location and password for the Stronghold
+  //
+  // Stronghold is an encrypted file that manages private keys.
+  // It implements best practices for security and is the recommended way of handling private keys.
+  let stronghold_path: PathBuf = "./example-strong.hodl".into();
+  let password: String = "my-password".to_owned();
+  let stronghold: Stronghold = Stronghold::new(&stronghold_path, password, None).await?;
+
+  // Alice creates and publishes their DID Document (see create_did and manipulate_did examples).
+  let mut builder: AccountBuilder = Account::builder().storage(stronghold);
+  let mut alice_account: Account = builder.create_identity(IdentitySetup::default()).await?;
+  alice_account
+    .update_identity()
+    .create_method()
+    .content(MethodContent::GenerateX25519)
+    .fragment("kex-0")
+    .scope(MethodScope::key_agreement())
+    .apply()
+    .await?;
+
+  // Bob creates and publishes their DID Document (see create_did and manipulate_did examples).
+  let mut bob_account: Account = builder.create_identity(IdentitySetup::default()).await?;
+  bob_account
+    .update_identity()
+    .create_method()
+    .content(MethodContent::GenerateX25519)
+    .fragment("kex-0")
+    .scope(MethodScope::key_agreement())
+    .apply()
+    .await?;
+
+  // Alice and Bob tell each other their DIDs. They each resolve the DID Document of the other
+  // to obtain their X25519 public key. Note that in practice, they would run this code completely
+  // separately.
+  let client: Client = Client::new().await?;
+
+  // Alice: resolves Bob's DID Document and extracts their public key.
+  let bob_document: ResolvedIotaDocument = client.resolve(bob_account.did()).await?;
+  let bob_method: &IotaVerificationMethod = bob_document
+    .document
+    .resolve_method("kex-0", Some(MethodScope::key_agreement()))
+    .unwrap();
+  let bob_public_key: Vec<u8> = bob_method.data().try_decode()?;
+
+  // Bob: resolves Alice's DID Document and extracts their public key.
+  let alice_document: ResolvedIotaDocument = client.resolve(alice_account.did()).await?;
+  let alice_method: &IotaVerificationMethod = alice_document
+    .document
+    .resolve_method("kex-0", Some(MethodScope::key_agreement()))
+    .unwrap();
+  let alice_public_key: Vec<u8> = alice_method.data().try_decode()?;
+
+  // Alice encrypts the data using Diffie-Hellman key exchange
+  let agreement: AgreementInfo = AgreementInfo::new(
+    b"apu".to_vec(),
+    b"apv".to_vec(),
+    b"pub_info".to_vec(),
+    b"priv_info".to_vec(),
+  );
+  let encryption_options: EncryptionOptions =
+    EncryptionOptions::new(EncryptionAlgorithm::Aes256Gcm, CEKAlgorithm::ECDH_ES { agreement });
+  let message: &[u8] = b"This msg will be encrypted and decrypted";
+  let encrypted_data: EncryptedData = alice_account
+    .encrypt_data(
+      message,
+      b"associated_data",
+      &encryption_options,
+      "kex-0",
+      bob_public_key.into(),
+    )
+    .await?;
```

Comment:
Just to confirm: I can't point to a specific RFC section, but I'm pretty sure the ephemeral key pair can be discarded afterwards. The sender has no use for the `ephemeral_secret` after the message was sent.

Reply:
> I'm actually uncertain whether Alice needs to retain said ephemeral X25519 keypair or if it is discarded and a new one generated for each anonymous-encrypted message they send? 

[Section 5.6.3.3 of NIST Special Publication 800-56A Revision 2](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-56Ar2.pdf) states the following: 

> An ephemeral private key shall be used in exactly one key-establishment transaction,
with one exception: an ephemeral private key may be used in multiple DLC key-transport
transactions that are transporting identical secret keying material simultaneously (or
within a short period of time; see the broadcast scenario in Section 7). **After its use, an
ephemeral private key shall be destroyed.**

However note that the NIST recommendation also states: 

> If a recipient of an ephemeral public key requires assurance that the claimed owner of
that public key has possession of the corresponding private key, then, in order to obtain
that assurance in compliance with this Recommendation, such assurance shall be
obtained as specified in Section 5.6.2.2.4. 

In the case of DID Comm I don't think it makes sense for Alice to retain said ephemeral key as those messages are assumed to be asynchronous, but for an ordinary key exchange it could be nice for the Account to provide a way for Alice to hold on to the ephemeral key in case Bob asks Alice to prove ownership of the corresponding private key.